### ISSUE TITLE
split-pages doesn't retain pdf version of original

### ISSUE THREAD

**Issue by matthewlenz** [qpdf/qpdf#610]
$ qpdf --version
qpdf version 10.5.0

I don't know if this is a feature or a bug.  Does qpdf inspect the content and set it to the "real" minimum version (if so, bravo) instead of whatever the original renderer set it at?

I had a couple dozen or so docs of different versions:

```
$ for i in $(ls *.pdf); do pdfinfo $i 2>/dev/null | grep 'PDF version'; done | sort | uniq -c
      2 PDF version:     1.3
     16 PDF version:     1.4
      1 PDF version:     1.5
      4 PDF version:     1.7
```

I split them out using:

`$ for i in $(ls *.pdf); do qpdf --no-warn --split-pages $i pages/$(basename $i .pdf)_%d.pdf; done`

Then I examined the resulting split pages and each was set to version 1.3:

```
$ for i in $(ls *.pdf); do pdfinfo $i 2>/dev/null | grep 'PDF version'; done | sort | uniq -c
    371 PDF version:     1.3
```

**Comment by jberkenbilt** [comment]
This is probably a bug. I wonder whether `--pages` suffers from the same problem. I will make sure this is fixed in 10.6 or whatever is next. Thanks.

**Comment by jberkenbilt** [comment]
This bug has been in qpdf since the beginning. I'm kind of shocked no one has ever noticed it before and that it never occurred to me. It seems that the correct behavior is to take the highest version across all the files that contribute to the output and use that as the output version. Right now, qpdf just takes the version from the primary input file, which, if `--empty`, is 1.3. Since `--split-pages` uses `--empty` internally, `--split-pages` always ends up at 1.3, which is definitely wrong. I will fix it today so the fix will be in 10.6.

**Comment by jberkenbilt** [comment]
> I will fix it today so the fix will be in 10.6.

where today = sometime in the next few days :-)
